src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -0,0 +1,401 @@
+import "@polymer/paper-icon-button/paper-icon-button";
+import { Layer, Marker, Circle, Map } from "leaflet";
+import {
+  LitElement,
+  TemplateResult,
+  css,
+  html,
+  property,
+  PropertyValues,
+  CSSResult,
+} from "lit-element";
+
+import "../../map/ha-entity-marker";
+
+import {
+  setupLeafletMap,
+  LeafletModuleType,
+} from "../../../common/dom/setup-leaflet-map";
+import computeStateDomain from "../../../common/entity/compute_state_domain";
+import computeStateName from "../../../common/entity/compute_state_name";
+import debounce from "../../../common/util/debounce";
+import parseAspectRatio from "../../../common/util/parse-aspect-ratio";
+import { HomeAssistant } from "../../../types";
+import computeDomain from "../../../common/entity/compute_domain";
+import { LovelaceCard } from "../types";
+import { LovelaceCardConfig } from "../../../data/lovelace";
+import { EntityConfig } from "../entity-rows/types";
+import { processConfigEntities } from "../common/process-config-entities";
+
+export interface MapCardConfig extends LovelaceCardConfig {
+  title: string;
+  aspect_ratio: string;
+  default_zoom?: number;
+  entities?: Array<EntityConfig | string>;
+  geo_location_sources?: string[];
+}
+
+class HuiMapCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement() {
+    await import(/* webpackChunkName: "hui-map-card-editor" */ "../editor/config-elements/hui-map-card-editor");
+    return document.createElement("hui-map-card-editor");
+  }
+
+  public static getStubConfig() {
+    return { entities: [] };
+  }
+
+  @property() public hass?: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true })
+  public isPanel = false;
+
+  @property()
```

Comment:
Can we decide on a standard for decorators? I don't like how we're doing inline sometimes but not always. I would vote for not inline as it is easier to be consistent and avoid long lines

Reply:
Well, prettier will put it on a new line if the property options get too long (as happened on the prop above this one).